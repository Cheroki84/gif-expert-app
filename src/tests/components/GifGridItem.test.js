import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas sobre el archivo GifGridItem', () => {
    
    const title = 'Un titulo'
    const url = 'https://mipagina.com'
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe de mostrar el componente correctamente', () => {
        
    expect(wrapper).toMatchSnapshot();

    });

    test('El párrafo debe de contener únicamente el title', () => {

        const p = wrapper.find('p');

        expect(p.text()).toBe(title);
        
    });

    test('El src y el alt de la imagen debe ser igual a las props', () => {
        
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
    
    test('Debe de tener la clase animate__bounce', () => {
        const div = wrapper.find('div');

        expect(div.prop('className').includes('animate__bounce')).toBe(true);

    });
    
});
