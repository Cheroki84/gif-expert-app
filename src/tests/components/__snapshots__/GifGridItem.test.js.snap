// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el archivo GifGridItem Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce "
>
  <img
    alt="Un titulo"
    src="https://mipagina.com"
  />
  <p>
    Un titulo
  </p>
</div>
`;
